src/sagemaker/estimator.py

```diff
@@ -570,7 +571,9 @@ def _ensure_base_job_name(self):
         """Set ``self.base_job_name`` if it is not set already."""
         # honor supplied base_job_name or generate it
         if self.base_job_name is None:
-            self.base_job_name = base_name_from_image(self.training_image_uri())
+            self.base_job_name = get_jumpstart_base_name_if_jumpstart_model(
```

Comment:
non-blocking: I think the implementation of line 1215 is more readable, so to be consistent I would prefer:
```py
jumpstart_base_name = get_jumpstart_base_name_if_jumpstart_model(...)
self.base_job_bame = jumpstart_base_name or base_name_from_image(self.training_image_uri())
```